### PR TITLE
[New Rule] Azure Event Hub Authorization Rule Created or Updated

### LINKED ISSUE #140: [New Rule] Azure Create or Update Event Hub Authorization Rule
## Description
Identifies when an Event Hub Authorization Rule is created or updated in Azure. An authorization rule is associated with specific rights, and carries a pair of cryptographic keys. When you create an Event Hubs namespace, a policy rule named `RootManageSharedAccessKey` is created for the namespace. This has manage permissions for the entire namespace and it's recommended that you treat this rule like an administrative root account and don’t use it in your application.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`azure`

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-shared-access-signature

## Syntax
```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:MICROSOFT.EVENTHUB/NAMESPACES/AUTHORIZATIONRULES/WRITE and event.outcome:Success
```

## Example Data
image

### PR DESCRIPTION
**Pull request by bm11100**
## Issue(s)
resolves #140 

## Description

Identifies when an Event Hub Authorization Rule is created or updated in Azure. An authorization rule is associated with specific rights, and carries a pair of cryptographic keys. When you create an Event Hubs namespace, a policy rule named RootManageSharedAccessKey is created for the namespace. This has manage permissions for the entire namespace and it's recommended that you treat this rule like an administrative root account and don't use it in your application.

## Checklists
Use ~~strikethroughs~~ to remove items which are not applicable to this issue.

### For submitter
- [x] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
  -- Query field process.name.text doesn't currently exist in endgame-* eventing data, but does exist for winlogbeats, and is case insensitive.
- [x] Verify all query fields co-exist within a single ECS version and annotated the minimum version
  -- 1.4.0
- [ ] Verified the query detects the intended event(s)
  - detonate
  - search in _discover_
  - trigger as a custom signal
- [x] Create rule using `create-rule`
- [x] Verify and convert to standard linting (`toml-lint -f <rule-file>`)
- [x] Run tests using `pytest -x -v unit_tests` or `make test`
- [ ] Internal search to determine noise and FP

### For reviewers
- [ ] Verify existing rule for activity doesn't exist as a siem or endpoint rule (unless intentionally specified in issue)
- [ ] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
- [ ] Verify all query fields co-exist within a single ECS version and annotated the minimum version
- [ ] Internal search to determine noise and FP
- [ ] Verify metadata accuracy and spelling

### CODE REVIEW

File: rules/azure/collection_update_event_hub_auth_rule.toml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    username, hostname, and/or resource name should be making changes in your environment. Authorization rule additions
+    or modifications from unfamiliar users or hosts should be investigated. If known behavior is causing false
+    positives, it can be exempted from the rule.
+    """,
+]
+from = "now-25m"
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Azure Create or Update Event Hub Authorization Rule"
```
Comment (brokensound77) [inline rules/azure/collection_update_event_hub_auth_rule.toml:27]:
```suggestion
name = "Azure Event Hub Authorization Rule Created or Updated"
```

File: rules/azure/collection_update_event_hub_auth_rule.toml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+[metadata]
+creation_date = "2020/08/18"
+ecs_version = ["1.5.0"]
```
Comment (brokensound77) [inline rules/azure/collection_update_event_hub_auth_rule.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

Comment (brokensound77) [inline rules/azure/collection_update_event_hub_auth_rule.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

**Review by threat-punter — APPROVED** [review]
✅

**Pull request merged**